### PR TITLE
Add register_stable_asset and update_stable_asset

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #1723

### CODE REVIEW

**Review by xlc — APPROVED** [review]
I am not sure if we want to maintain `NextStableAssetId`  here. It could be in the stable asset pallet.

But I guess it will make everything easier to do this way so I am fine with that.

File: modules/asset-registry/src/lib.rs [hunk 1]
```diff
@@ -244,41 +327,76 @@ impl<T: Config> Pallet<T> {
...
+			let old_multi_locations = maybe_multi_locations.as_mut().ok_or(Error::<T>::AssetIdNotExists)?;
+
+			AssetMetadatas::<T>::try_mutate(
+				AssetIds::ForeignAssetId(foreign_asset_id),
+				|maybe_asset_metadatas| -> DispatchResult {
+					ensure!(maybe_asset_metadatas.is_some(), Error::<T>::AssetIdNotExists);
+
+					// modify location
+					if location != old_multi_locations {
+						LocationToCurrencyIds::<T>::remove(old_multi_locations.clone());
```
Comment (xlc) [inline modules/asset-registry/src/lib.rs:339]:
```suggestion
						LocationToCurrencyIds::<T>::remove(&old_multi_locations);
```

Comment (zjb0807) [reply modules/asset-registry/src/lib.rs:339]: It was used here.
```
*old_multi_locations = location.clone();
```

**Comment by zjb0807** [comment]
> I am not sure if we want to maintain `NextStableAssetId` here. It could be in the stable asset pallet.
> 

Maintaining NextStableAssetId here is really bad. But ed will be checked during `mint`, if it is not registered, `mint` will fail, so it will be forced to be registered.

**Pull request merged**